tree-sitter-stack-graphs/src/lib.rs

```diff
@@ -0,0 +1,620 @@
+//! }
+//! ```
+//!
+//! A _push scoped symbol_ node requires a `scope` attribute.  Its value must be a reference to an `exported`
+//! node that you've already created. (This is the exported scope node that will be pushed onto the scope
+//! stack.)  For instance:
+//!
+//! ``` skip
+//! (identifier) @id {
+//!   node new_exported_scope_node
+//!   attr (new_exported_scope_node) is_exported
+//!   node new_push_scoped_symbol_node
+//!   attr (new_push_scoped_symbol_node)
+//!     type = "push_scoped_symbol",
+//!     symbol = (source-text @id),
+//!     scope = new_exported_scope_node
+//! }
+//! ```
+//!
+//! Nodes of type `scope` allow an optional `is_exported` attribute, that is required to use the scope
+//! in a `push_scoped_symbol` node.
+//!
+//!
+//! ### Annotating nodes with location information
+//!
+//! You can annotate any stack graph node that you create with location information, identifying
+//! the portion of the source file that the node "belongs to".  This is _required_ for definition
+//! and reference nodes, since the location information determines which parts of the source file
+//! the user can _click on_, and the _destination_ of any code navigation queries the user makes.
+//! To do this, add a `source_node` attribute, whose value is a syntax node capture:
+//!
+//! ``` skip
+//! (function_definition name: (identifier) @id) @func {
+//!   node def
+//!   attr (def) type = "pop_symbol", symbol = (source-text @id), source_node = @func, is_definition
+//! }
+//! ```
+//!
+//! Note how in this example, we use a different syntax node for the _target_ of the definition
+//! (the entirety of the function definition) and for the _name_ of the definition (the content of
+//! the function's `name`).
+//!
+//! ### Connecting stack graph nodes with edges
+//!
+//! To connect two stack graph nodes, use the `edge` statement to add an edge between them:
+//!
+//! ``` skip
+//! (function_definition name: (identifier) @id) @func {
+//!   node def
+//!   attr (def) type = "pop_symbol", symbol = (source-text @id), source_node = @func, is_definition
+//!   node body
+//!   edge def -> body
+//! }
+//! ```
+//!
+//! To implement shadowing (which determines which definitions are selected when multiple are available),
+//! you can add a `precedence` attribute to each edge to indicate which paths are prioritized:
+//!
+//! ``` skip
+//! (function_definition name: (identifier) @id) @func {
+//!   node def
+//!   attr (def) type = "pop_symbol", symbol = (source-text @id), source_node = @func, is_definition
+//!   node body
+//!   edge def -> body
+//!   attr (def -> body) precedence = 1
+//! }
+//! ```
+//!
+//! (If you don't specify a `precedence`, the default is 0.)
+//!
+//! ### Referring to the singleton nodes
+//!
+//! The _root node_ and _jump to scope node_ are singleton nodes that always exist for all stack
+//! graphs.  You can refer to them using the `ROOT_NODE` and `JUMP_TO_SCOPE_NODE` global variables:
```

Comment:
Even though the section on global variables in `tree-sitter-graph`'s language reference describes declaring the global variable,  what do you think about including the `global ROOT_NODE` statement as part of this example stanza?